src/zenml/steps/entrypoint_function_utils.py

```diff
@@ -235,17 +236,40 @@ def _validate_input_value(
             parameter: The function parameter for which the value was provided.
             value: The input value.
         """
-        config_dict = ConfigDict(arbitrary_types_allowed=False)
-
-        # Create a pydantic model with just a single required field with the
-        # type annotation of the parameter to verify the input type including
-        # pydantics type coercion
-        validation_model_class = create_model(
-            "input_validation_model",
-            __config__=config_dict,
-            value=(parameter.annotation, ...),
-        )
-        validation_model_class(value=value)
+        annotation = parameter.annotation
+
+        # Handle Optional types
+        origin = get_origin(annotation)
+        if origin is Union:
```

Comment:
Is there no better way to leverage pydantic for this case? This seems like very messy code, and I'm not even sure why your example failed when using pydantic